stan/math/prim/mat/fun/to_matrix.hpp

```diff
@@ -2,25 +2,53 @@
 #define STAN_MATH_PRIM_MAT_FUN_TO_MATRIX_HPP
 
 #include <stan/math/prim/mat/fun/Eigen.hpp>
- // stan::scalar_type
 #include <vector>
 
 namespace stan {
   namespace math {
-
-    // matrix to_matrix(matrix)
-    // matrix to_matrix(vector)
-    // matrix to_matrix(row_vector)
+    /**
+     * Returns the matrix itself.
+     *
+     * @tparam T type of the scalar
+     * @tparam R number of rows
+     * @tparam C number of columns
+     * @param matrix matrix
+     */
     template <typename T, int R, int C>
     inline Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>
     to_matrix(Eigen::Matrix<T, R, C> matrix) {
       return matrix;
     }
 
-    // matrix to_matrix(real[, ])
+    /**
+     * Returns a matrix representation of the vector in column-major order
+     * with the specified number of rows and columns.
+     *
+     * @tparam T type of the scalar
+     * @param vec vector of values
+     * @param m rows
+     * @param n columns
+     * @throw <code>std::invalid_argument</code> if the sizes do not match
+     */
+    template <typename T>
+    inline Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>
+    to_matrix(const std::vector<T>& vec, int m, int n) {
+      static const char* fun = "to_matrix(array)";
+      check_size_match(fun, "rows * columns", m * n, "vector size", vec.size());
+      return Eigen::Map<const Eigen::Matrix<T, Eigen::Dynamic,
+                                            Eigen::Dynamic> >(&vec[0], m, n);
+    }
+
+    /**
+     * Returns a matrix representation of the 2d vector with the same dimensions
```

Comment:
`2d vector` should be `standard vector of standard vectors`.  We want to tease apart standard vectors and Eigen vectors, and in general want to use standard English spelling and punctuation in our doc.